### PR TITLE
Format CSE output as a json object

### PR DESCRIPTION
**Pull request by bowang-666**
Format CSE output as a json object so that RP can better consumes it.

### CODE REVIEW

File: pkg/agent/const.go [hunk 1]
```diff
@@ -73,6 +73,7 @@ const (
 // cloud-init (i.e. ARM customData) source file references
 const (
 	kubernetesNodeCustomDataYaml  = "linux/cloud-init/nodecustomdata.yml"
+	kubernetesCSEStartScript      = "linux/cloud-init/artifacts/cse_start.sh"
 	kubernetesCSECommandString    = "linux/cloud-init/artifacts/cse_cmd.sh"
```
Comment (xuto2) [inline pkg/agent/const.go:77]: super nit: can we switch the order of L76 and L77? I'd like to see them in a logical order

**Review by xuto2 — APPROVED** [review]

File: parts/linux/cloud-init/artifacts/cse_start.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1
+EXIT_CODE=$?
+systemctl --no-pager -l status kubelet >> /var/log/azure/cluster-provision-cse-output.log 2>&1
+OUTPUT=$(cat /var/log/azure/cluster-provision-cse-output.log | head -n 30)
+JSON_STRING=$( jq -n \
+                  --arg ec "$EXIT_CODE" \
+                  --arg op "$OUTPUT" \
+                  --arg er "" \
```
Comment (zjpjack) [inline parts/linux/cloud-init/artifacts/cse_start.sh:8]: I don't get this part, we always make er empty?

Comment (bowang-666) [reply parts/linux/cloud-init/artifacts/cse_start.sh:8]: currently we don't have a very good way to get error message from error code, but in future we will add detailed error message here.

File: pkg/agent/const.go (hunk 1 above)
Comment (bowang-666) [reply pkg/agent/const.go:76]: fixed

File: parts/linux/cloud-init/artifacts/cse_start.sh (hunk 2 above)
Comment (bowang-666) [reply parts/linux/cloud-init/artifacts/cse_start.sh:8]: this's a placeholder for now

**Review by zjpjack — APPROVED** [review]

**Pull request merged**